src/components/GameItem.js

```diff
@@ -1,32 +1,82 @@
-import React, { Component } from 'react';
-import '../App.css';
-import ItemIcons from '../ItemIcons.js';
-import PropTypes from 'prop-types';
+import React, { Component } from "react";
+import "../App.css";
+import ItemIcons from "../ItemIcons.js";
+import PropTypes from "prop-types";
 
 class GameItem extends Component {
+  constructor(props) {
+    super(props);
+    this.type = props.type;
+    this.state = {
+      spotted: false,
+      isLitter: false,
+      itemClassName: ""
+    };
+  }
+  // Method for what to do when an item is "spotted"
+  spotted = () => {
+    // Logic here
+    // If object is litter/has not been clicked before
+    if (this.props.type === "litter" && !this.props.spotted) {
+      this.props.clickCallback(this.props.type);
```

Comment:
You're using `this.props` to check the type (which is fine), but there is no prop for spotted, so when you say `this.props.spotted` it never changes from false.  You want to be checking `this.state.spotted`.